File: QEdata.js
doc = {
  t1: {
    A: "TEST",
    B: [
      {
        B1: "TEST",
        B2: "TEST",
      },
      {
        B1: "TEST",
        B2: "TEST",
      },
    ],
  },
};

doc1map = {// Not Working
  keyId: null,
  path: "t1.B.B1",
  bsonType: "string", 
};

doc2map = { // Working
  keyId: null,
  path: "t1.A",
  bsonType: "string",
  queries: {
    queryType: "equality" 
  },
};

File: QEdata.java
BsonDocument doc1 = new BsonDocument()
				.append("keyId", new BsonNull()))
				.append("path", new BsonString("t1.B.B1"))
				.append("bsonType", new BsonString("string")); // Not Working

BsonDocument doc2 = new BsonDocument()
				.append("keyId", new BsonNull())
				.append("path", new BsonString("t1.A")) // Working properly
				.append("bsonType", new BsonString("string"))
				.append("queries", new BsonDocument().append("queryType", new BsonString("equality")));`